### PR TITLE
Use RetroArch's system directory for Kickstart/BIOS files

### PR DESCRIPTION
**Pull request by Enverex**
To match the setup of other libretro modules, use the RA system folder for BIOS files rather than FS-UAE's default "Documents/FS-UAE/Kickstarts" folder.

### CODE REVIEW

**Comment by inactive123** [comment]
@repojohnray This OK with you if we merge this?

**Comment by repojohnray** [comment]
We have a lot of sub-directories under ~/FS-UAE :
   CD-ROMs  Cache   Configurations  Controllers  Data  Floppies  'Hard Drives'  Kickstarts  'Save States'  Themes

Moving only 'Kickstarts' today is not helpful.

**Comment by Enverex** [comment]
Themes obviously wouldn't be used, Save States can be updated to use the proper RetroArch folder too. The majority of the other folders would not be used though as they're typically used for the front-end aren't they? (which we aren't using).

**Cache** - Appears to be frontend only.
**Configurations** - This doesn't really need to exist, it's just where FS-UAE's frontend puts configs by default which the user could put anywhere. The core itself wouldn't be creating any with the exception of trying to read "Default.fs-uae" which could be put in a more appropriate RetroArch folder.
**Controllers** - This one is probably relevant so this will need to go somewhere.
**Data** - This is front-end only again by the looks of it.
**Floppies** - This is just another folder where people "can put their games" so it's not relevant.
**Hard Drives** - Same as floppies, this is just a "for your convenience folder" mainly used by the frontend.
**CD-ROMs** - Same as floppies and HDs.
**KickStarts** - This is what we're looking at moving now.
**Plugins** - Frontend again I believe.
**Save States** - Would be moved to the RetroArch savestates folder.
**Themes** - Frontend again.

So basically the only ones (4) we need are Configurations (for Default), Controllers, KickStarts and Save States. Two of which map directly to existing RetroArch folders. The latter two could probably go into a sub-folder of the system folder.

**Comment by andres-asm** [comment]
Savestates are handled by the frontend.
If the core doesn't support serialization then it can use the save dir

**Comment by repojohnray** [comment]
You are right; We may begin the transition. In my opinion using: #ifdef USE_FSUAEDIRSTRUCTURE ... to keep the usual FS-UAE directory structure too is the right way to proceed.

**Comment by andres-asm** [comment]
I think you should use a fs-uae dir in RETRO_SAVE_DIRECTORY for everythin
with subfolders

**Comment by Enverex** [comment]
If that's the consensus, I'll close this pull and do it using the RetroArch save folder + subfolders and a switch to make the existing stand-alone FS-UAE structure optional?

**Comment by repojohnray** [comment]
OK - We may proceed.
